test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -603,4 +610,146 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
 			}, util.Timeout, util.Interval).Should(testing.BeNotFoundError())
 		})
 	})
+
+	ginkgo.When("Reconciling clusterQueue pending workload status", func() {
+		var (
+			clusterQueue   *kueue.ClusterQueue
+			localQueue     *kueue.LocalQueue
+			onDemandFlavor *kueue.ResourceFlavor
+		)
+
+		ginkgo.BeforeEach(func() {
+			onDemandFlavor = testing.MakeResourceFlavor(flavorOnDemand).Obj()
+			gomega.Expect(k8sClient.Create(ctx, onDemandFlavor)).To(gomega.Succeed())
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(flavorOnDemand).
+						Resource(corev1.ResourceCPU, "5", "5").Obj(),
+				).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
+		})
+
+		ginkgo.It("Should update of the pending workloads when a new workload is scheduled", func() {
+			workloads := []*kueue.Workload{
+				testing.MakeWorkload("one", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "2").Request(resourceGPU, "2").Obj(),
+				testing.MakeWorkload("two", ns.Name).Queue(localQueue.Name).
+					Request(corev1.ResourceCPU, "3").Request(resourceGPU, "3").Obj(),
+			}
+
+			ginkgo.By("Creating workloads")
```

Comment:
Before we create any workloads add an assert to check that the status is empty (no pending workloads)

Reply:
check added